process.py

```diff
@@ -345,20 +379,25 @@ def print_ASCTB_table():
                             exit_with_error(error)
 
                     cell_feature = features[cell]
-                    record = anatomical_structure + get_data_block(cell_feature) + "\t"
-
-                    # add cell-specific features
-                    add_features(record, cell_feature)
+                    cellfeatureDataBlockList=get_data_block(cell_feature)
+                    for cellData in cellfeatureDataBlockList:
+                        anatomical_structure_safe.append(cellData)   
+                    #add cell-specific features
+                    add_features(anatomical_structure_safe, cell_feature)
                     node_output = True
+                    anatomical_structure_safe=anatomical_structure[:]
 
             # if node is leaf and doesn't have any assigned cells or
             # features then we output it here.
+            ###CHANGES NEED TO BE DONE
             if not node_output:
-                record = anatomical_structure + "\t\t\t\t\t" * (1 + max_genes + max_proteins + max_proteoforms + max_lipids + max_metabolites + max_ftu + max_references) + "\n"
-                out_file.write(record)
-
+                anatomical_structure.append("" * (1 + max_genes + max_proteins + max_proteoforms + max_lipids + max_metabolites + max_ftu + max_references))
+                for anatomy in anatomical_structure:
+                    for structure in anatomy:
+                        record.append(structure)
+                csvwriter.writerow(record)
```

Comment:
Rather than writing rows here, return an array of records (array of strings)  [or yield records]. csvwriter shouldn't be global.